test/ci.py

```diff
@@ -0,0 +1,187 @@
+#
+# Copyright 2019-2019 Ghent University
+#
+# This file is part of vsc-install,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# the Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/hpcugent/vsc-install
+#
+# vsc-install is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Library General Public License as
+# published by the Free Software Foundation, either version 2 of
+# the License, or (at your option) any later version.
+#
+# vsc-install is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Library General Public License for more details.
+#
+# You should have received a copy of the GNU Library General Public License
+# along with vsc-install. If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Test CI functionality
+
+@author: Kenneth Hoste (Ghent University)
+"""
+import logging
+import os
+import re
+import shutil
+import sys
+import tempfile
+
+try:
+    from cStringIO import StringIO  # Python 2
+except ImportError:
+    from io import StringIO  # Python 3
+
+import vsc.install.ci
+from vsc.install.ci import JENKINSFILE_REVISION, TOX_INI_REVISION, gen_jenkinsfile, gen_tox_ini, write_file
+from vsc.install.testing import TestCase
+
+
+def read_file(path):
+    """Read file at specified path, and return contents."""
+    with open(path, 'r') as handle:
+        return handle.read()
+
+
+class CITest(TestCase):
+    """License related tests"""
+
+    def setUp(self):
+        """Test setup"""
+        super(CITest, self).setUp()
+
+        self.cwd = os.getcwd()
+        self.tmpdir = tempfile.mkdtemp()
+        self.tmpdir_name = os.path.basename(self.tmpdir)
+
+    def tearDown(self):
+        """Test cleanup"""
+        shutil.rmtree(self.tmpdir)
+        os.chdir(self.cwd)
+
+        super(CITest, self).tearDown()
+
+    def run_function(self, function, *args, **kwargs):
```

Comment:
wtf is this? generate the expected text and compare it with a static file in test/data
if you want to test `write_file`, just use mock?

Reply:
This is mostly about capturing the logging to stdout (the stuff the user gets to see, at least)

Reply:
euh, not sure what he added value of unittesting the log output is, but whatever i guess

Reply:
removed